File: src/main/java/com/example/dpop/orchestrator/api/v1/authentication/sms/UseStep.java
package com.example.dpop.orchestrator.api.v1.authentication.sms;

import com.example.dpop.auth_sms.EnrollmentRef;

import java.util.List;

public sealed interface UseStep {
    record NeedInput(List<String> missingFields) implements UseStep {}
    record VerifyChallenge(EnrollmentRef ref, String tan) implements UseStep {}
}

File: src/main/kotlin/com/example/dpop/orchestrator/api/v1/authentication/sms/UseStep.kt
package com.example.dpop.orchestrator.api.v1.authentication.sms

import com.example.dpop.auth_sms.EnrollmentRef

sealed interface UseStep {
    @JvmRecord
    data class NeedInput(val missingFields: List<String>) : UseStep

    @JvmRecord
    data class VerifyChallenge(val ref: EnrollmentRef, val tan: String?) : UseStep
}
